ddtrace/profiling/_threading.pyx

```diff
@@ -108,28 +119,22 @@ class _ThreadLink(_thread_link_base):
     ):
         # type: (...) -> None
         """Link an object to the current running thread."""
-        self._thread_id_to_object[_thread.get_ident()] = weakref.ref(obj)
+        # Because threads might become tasks with some frameworks (e.g. gevent),
+        # we retrieve the thread ID using the C API instead of the Python API.
+        self._thread_id_to_object[<object>PyLong_FromLong(PyThreadState_Get().thread_id)] = weakref.ref(obj)
```

Comment:
If my previous comment was correct, then I think the keys to `_thread_id_to_object` are actually pointers.  This would be easy to verify by printing the numerical value of `thread_id`.  If it's a pointer, it'll be a moderately large value.  If it's a true thread_id, then it should be small (especially if the test is run in a container running on a PID namespace).

Reply:
I don't know enough about the dynamical operation of Python interpreter to figure out whether my observation matters.  I think it does because the interpreter keeps these threads as a linked list, so as long as threads have differently-scaled lifetimes then `next` for a given thread will change much more rapidly than its own ID.  I can see how this might not matter for some tests though.